### DISCUSSION TITLE
Vercel deploys each and every branch

### DISCUSSION THREAD

**Discussion by yousefelgoharyx** [vercel/community#3213] [Help] [answered]
### Problem Description

I don't want Vercel to deploy every branch I have, that's really annoying and it pollutes the deployment history. I want to select specific branches that Vercel can deploy.
Is there a way to do that?

**Comment by 2hmad** [comment]


**Reply by yousefelgoharyx** [reply]
What If need to deploy more than 1 branch

**Comment by 2hmad** [comment, marked as answer]
If you need to deploy the production environment only you can change your project configuration on vercel 
 **Settings -> Git -> Ignored Build Step**
then, change it to `only build production`

**Reply by amyegan** [reply]
If you need a more customized workflow, you can write your own script to determine which branches to build or ignore. Some examples can be found here: https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel

To deploy only from `main` and `staging` branches, for example, you could a bash script for Ignored Build Step set to
```bash
# script.sh 

echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"

if [[ "$VERCEL_GIT_COMMIT_REF" == "staging" || "$VERCEL_GIT_COMMIT_REF" == "main"  ]] ; then
  # Proceed with the build
    echo "✅ - Build can proceed"
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```
